docs/proposals/21-ipv6-singlestack-local.md

```diff
@@ -0,0 +1,213 @@
+The feature gate cannot be enabled if the `ReversedVPN` feature gate is disabled (the legacy VPN tunnel solution is [not supported by provider-local](https://github.com/gardener/gardener/blob/83de074f1bc1c009f92e97a08289340591377af6/docs/extensions/provider-local.md#current-limitations) anyway).
+
+### `Shoot` API
+
+The `Shoot.spec.networking` section is extended as follows:
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Shoot
+spec:
+  networking:
+    # ...
+    ipFamilies:
+    - IPv6
+    podsV6: dead:.../64
+    nodesV6: dead:.../64
+    servicesV6: dead:.../64
+```
+
+`ipFamilies` is the central setting for specifying the IP families used for shoot networking.
+This field is inspired by the `Service.spec.ipFamilies` field in Kubernetes ([doc](https://kubernetes.io/docs/concepts/services-networking/dual-stack/#services)).
+The only valid value and default is `["IPv4"]` (IPv4 single-stack).
+If the `IPv6SingleStack` feature gate is enabled, `["IPv6"]` can be specified to switch to IPv6 single-stack.
+Later on, `["IPv4","IPv6"]` or `["IPv6","IPv4"]` can be supported for dual-stack networking (ordering is relevant for Kubernetes configuration like the `--service-cluster-ip-range` flag).
+
+Additionally, IPv6-equivalents of all CIDR and mask fields are introduced, e.g., `Shoot.spec.networking.{pods,services,nodes}V6` and `Shoot.spec.kubernetes.kubeControllerManager.nodeCIDRMaskSizeV6`.
+Depending on the `ipFamilies` value, either the IPv4 field or the IPv6 field may be specified – or both in a potential dual-stack implementation.
+
+All gardener components and extensions need to respect the `ipFamilies` field and handle it correctly, e.g., in API validation, defaulting, and configuring Shoot components.
+
+### `Seed` API
+
+Similar to the `Shoot` API, the `Seed.spec.networks` section is extended as follows:
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Seed
+spec:
+  networks:
+    ipFamilies:
+    - IPv6
+    nodesV6: dead:.../64
+    podsV6: dead:.../64
+    servicesV6: dead:.../64
+    shootDefaults:
+      podsV6: dead:.../64
+      servicesV6: dead:.../64
+```
+
+`ipFamilies` has the same semantics as `Shoot.spec.networking.ipFamilies`.
+Again, IPv6-equivalents of all CIDR and mask fields are introduced, e.g. `Seed.spec.networks.{nodes,pods,services}V6` and `Seed.spec.networks.shootDefaults.{pods,services}V6`
+Depending on the `ipFamilies` value, either the IPv4 field or the IPv6 field may be specified – or both in a potential dual-stack implementation.
+
+The existing `Seed.spec.networks.blockCIDRs` field is augmented to allow IPv6 CIDR values in addition to IPv4 values.
+Validation for this field is laxer and CIDRs of both IP families are allowed independently of the `ipFamilies` setting.
+Values of the `blockCIDRs` field are only relevant for excluding blocks in `NetworkPolicies` and hence it doesn't hurt to add CIDR blocks that are not reachable anyway.
+
```

Comment:
We should keep in mind that GCP ans AWS are allocating and routing IPv6 space to VMs.
To make this PR future proof, we need to keep in mind that instead of IPs, podsV6 and nodesV6 also have to support keyword like "reqestGUA" or "requestULA"

Reply:
Thanks, we have this in mind.
This will definitely be considered in the next steps after implementing this GEP, i.e., when bringing IPv6 to cloud infrastructure.
As described in the motivational section, we focus on the local environment to explicitly disregard these discussions for now.